hcl2template/types.packer_config.go

```diff
@@ -280,6 +285,99 @@ func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceBlock, blocks [
 	return res, diags
 }
 
+func (cfg *PackerConfig) HCL2ProvisionerPrepare(typeName string, data map[string]interface{}) (packer.Provisioner, hcl.Diagnostics) {
+	// This will interpolate build variables by decoding the provisioner block again
+	var diags hcl.Diagnostics
+	if data == nil {
+		diags = append(diags, &hcl.Diagnostic{
+			Summary: fmt.Sprintf("failed loading %s", typeName),
+			Detail:  "unable to prepare provisioner with build variables interpolation",
+		})
+		return nil, diags
+	}
```

Comment:
Is it an issue when no data was passed ?

Reply:
This can be silent, then we just return the provisioner. No interpolation will happen. A warning telling that any build variable won't be interpolated maybe?